system/include/emscripten/em_asm.h

```diff
@@ -0,0 +1,84 @@
+#ifndef __em_asm_h__
+#define __em_asm_h__
+
+
+#define CONCAT(x, y) CONCAT_(x, y)
```

Comment:
The C preprocessor is... weird. This is one of those cases where we solve problems via layers of indirection. By deferring this twice, we can concatenate the result of another macro.

So,
```c
FOO(a)##FOO(b);
CONCAT_(FOO(a), FOO(b));
CONCAT(FOO(a), FOO(b));
```

becomes
```c
a_foo##b_foo;
FOO(a)FOO(b); // which emits a preprocessor error then expands to `a_foo b_foo`
a_foob_foo;
```

This entire PR is more or less just confounding the preprocessor into doing what we want.